subprojects/core/src/main/java/org/gradle/internal/service/scopes/ExecutionGlobalServices.java

```diff
@@ -89,6 +90,10 @@ PropertyAnnotationHandler createInternalAnnotationHandler() {
         return new NoOpPropertyAnnotationHandler(Internal.class);
     }
 
+    PropertyAnnotationHandler createReplacedByAnnotationHandler() {
```

Comment:
Can you also update 'ValidateTaskPropertiesIntegrationTest` to verify we don't complain about a property with this annotation.

Also, we should probably complain or fail when another input or output annotation is attached to a `@ReplacedBy` property, as the annotations should move to the new property.